### PR TITLE
WebUI: Sanitize user-input fields of PRs

### LINKED ISSUE #8203: Pull Requests: Sanitize to user provided fields
We should sanitize any user inputted fields in pull requests such as Title and Description to avoid potentially malicious attacks

### PR DESCRIPTION
**Pull request by itaigilo**
Closes #8203.

## Change Description

Adding sanitize for user-input fields of a PR, to prevent malicious attacks.
Currently it's only relevant for title + description when creating a PR.

Using the very popular [DOMPurify](https://github.com/cure53/DOMPurify) package.

### Testing Details

Verified manually that PR creation still works.

### CODE REVIEW

**Review by N-o-Z — APPROVED** [review]
Great thanks!
I feel like we should probably go over other UI components and verify we do the same for user inputs
Also - please don't close the related issue (or alternatively open a new one)
Since we should potentially do this for the backend as well

**Review by arielshaqed — CHANGES_REQUESTED** [review]
Not a fan.  For instance, this prevents me from having a PR title "`Fix <b> tags`":
- If I try to type it in, it helpfully purifies the first `<` and I get "`Fix &lt;b> tags`".
- If you set it on the server directly through the API and I try to read it from the GUI, I see "`Fix <b> tags</b>`".
Neither is correct.

In fact I expect React `{...}` to protect me!  So I would also expect React-Bootstrap components to inherit this behaviour.  This protection would would kinda protect _some other_ bad website, or some broken component in _our_ web UI, from failing, one that presumably runs React `dangerouslySetInnerHTML` or simply allows HTML injection.

Additionally, this enforces that I cannot _use the GUI_ to enter a string that we might fear.  But I can still use the API to enter the exact same string.

Basically, I believe that clients need to protect themselves when reading code, correctly.  Not sure this is the way ;-(
